Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java

```diff
@@ -602,13 +604,7 @@ public void onPlayerChangedWorldFlyReset(final PlayerChangedWorldEvent event) {
         final User user = ess.getUser(event.getPlayer());
 
         if (ess.getSettings().isWorldChangeFlyResetEnabled()) {
-            if (user.getBase().getGameMode() != GameMode.CREATIVE
-                    // COMPAT: String compare for 1.7.10
-                    && !user.getBase().getGameMode().name().equals("SPECTATOR")
-                    && !user.isAuthorized("essentials.fly")) {
```

Comment:
Shouldn't it be `essentials.fly.safelogin`?

Reply:
This is checked on line 336

https://github.com/EssentialsX/Essentials/blob/524340a49217668d9d7da127c0f5f4d5412fff61/Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java#L336

The fly permission is only checked after they do not have safe login perms.

Reply:
Ok, thanks for letting know